internal/controller/operatorconfigmap_controller.go

```diff
@@ -890,3 +784,79 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) reconcileSubscriptionLabelForWebhook() error {
+	sub, err := c.getSubscriptionByPackageName("ocs-client-operator")
+	if err != nil {
+		return err
+	}
+	if utils.AddLabel(sub, subscriptionLabelKey, subscriptionLabelValue) {
+		if err := c.update(sub); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) reconcileSubscriptions() error {
+	pkgNames := [][2]string{
+		// d/s and u/s names of packages for fallback
+		{"ocs-client-operator", ""},
+		{"odf-csi-addons-operator", "csi-addons"},
+		{"cephcsi-operator", "ceph-csi-operator"},
+		{"recipe", ""},
+	}
+	if c.getNoobaaSubManagementConfig() {
+		pkgNames = append(pkgNames, [2]string{"mcg-operator", "noobaa-operator"})
+	}
+	installPlansForApproval := map[string]struct{}{}
+	subsMissingInstallPlans := false
+	for _, pkgName := range pkgNames {
+		sub, err := c.getSubscriptionByPackageName(pkgName[0])
+		if pkgName[1] != "" && kerrors.IsNotFound(err) {
+			sub, err = c.getSubscriptionByPackageName(pkgName[1])
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
I don't see a need for such logic. If you find it update it, if not move to the next one. There is no need for hierarchy